File: Leads-Management-WebAPI/lead-management-webapi/Model/Lead.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace LeadManageAPI.Model
{
    public enum RentalStatus
    {
        LookingForRental,
        NotInterested,
        Rented,
        DoNotCall,
        Archived
    }

    public class Lead
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int Id { get; set; }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [EmailAddress]
        public string Email { get; set; }

        [Phone]
        public string Phone { get; set; }

        [Url]
        public string Source { get; set; }

        [JsonPropertyName("status")]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public RentalStatus Status { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime UpdatedDate { get; set; }

    }
}

File: Leads-Management-WebAPI/lead-management-client/src/app/lead.ts
export enum LeadStatus {
    LookingForRental = "Looking For Rental",
    NotInterested = "Not Interested",
    Rented = "Rented",
    DoNotCall = "Do Not Call",
    Archived = "Archived"
}

export interface ILeadItem {
    id: number,
    firstName: string,
    lastName: string,
    email: string,
    phone: string,
    source: string,
    status: LeadStatus,
    createdDate: Date,
    updatedDate: Date
}

export class LeadItem implements ILeadItem {
    constructor (){};
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    source: string;
    status: LeadStatus;
    createdDate: Date;
    updatedDate: Date;

    static fromItemJson (itemJson: LeadItemJSON){
        let leadItem = new LeadItem();
        leadItem.id = itemJson.id;
        leadItem.firstName = itemJson.firstName;
        leadItem.lastName = itemJson.lastName;
        leadItem.phone = itemJson.phone;
        leadItem.source = itemJson.source;
        leadItem.status = LeadStatus[itemJson.status];
        leadItem.createdDate = new Date(itemJson.createdDate);
        leadItem.updatedDate = new Date(itemJson.updatedDate);
        return leadItem;
    };
}

export interface LeadItemJSON {
    id: number,
    firstName: string,
    lastName: string,
    email: string,
    phone: string,
    source: string,
    status: string,
    createdDate: string,
    updatedDate: string
}